### DISCUSSION TITLE
API Documentation

### DISCUSSION THREAD

**Discussion by Manu5656** [GriefPrevention/GriefPrevention#1344] [General]
Hi, where can I find the GriefPrevention API documentation? I searched everywhere but couldn't find anything.

Thanks so much.

**Comment by Jikoo** [comment]
There isn't really any. There're [useful comments](https://github.com/TechFortress/GriefPrevention/blob/e6d608bcb9dc76572d077241e1d6da25ced19dd7/src/main/java/me/ryanhamshire/GriefPrevention/Claim.java#L127-L129) all over the code, but they're a poor substitute for Javadocs. Your best bet is to figure out what you need to know to do something, then figure out how GP handles a similar function.

For example, we want to make an addon that allows users to run certain commands only in admin claims: we need to know how to determine which claim (if any) a user is in and if a claim is an admin claim.
GP has commands that have behaviors based on whether or not the user is in a claim, so that's a start - we pop over to GP's [horrific command handling](https://github.com/TechFortress/GriefPrevention/blob/e6d608bcb9dc76572d077241e1d6da25ced19dd7/src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java#L967-L2922). There we identify a command that requires us to be in a claim and [see how the Claim is obtained](https://github.com/TechFortress/GriefPrevention/blob/e6d608bcb9dc76572d077241e1d6da25ced19dd7/src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java#L1146-L1148). The next lines make us aware that the claim will be null if no claim is present, so that's something to keep in mind.
Now we have a claim, we need to know if it's an admin claim. Fortunately, [Claim#isAdminClaim](https://github.com/TechFortress/GriefPrevention/blob/e6d608bcb9dc76572d077241e1d6da25ced19dd7/src/main/java/me/ryanhamshire/GriefPrevention/Claim.java#L96) exists, so that's pretty easy. Worst case scenario we could have taken a closer look at how administrative claims mode creates claims when a PlayerInteractEvent with the correct tool is fired.

It's definitely a bit of a pain, but I doubt you'll see a focus on documentation until GP v20 when addons are a real focus for behavior.
